pkg/util/cert_pool.go

```diff
@@ -54,7 +55,7 @@ func (cp *certPool) isCertificateDuplicate(certData []byte) bool {
 }
 
 // Append certificate to a pool
-func (cp *certPool) appendCertFromPEM(PEMdata []byte) error {
+func (cp *certPool) appendCertFromPEM(PEMdata []byte, filterExpiredCerts bool) error {
```

Comment:
suggestion: I can't imagine a situation where we'd need to configure `filterExpiredCerts` on a per-certificate basis like this.

Do you think it makes sense to add `filterExpiredCerts` as a boolean option on `certPool`, so we'd have `newCertPool(filterExpiredCerts bool) *certPool` instead and `appendCertFromPEM` would keep its current signature?

Reply:
You are right, that would be better, I will make the change later. Thanks!

Reply:
I have created new functions with a certPool receiver to preserve the signature of the previous function. I am not an expert in Go, so please let me know if you think of a better approach.